### DISCUSSION TITLE
Optimizing CSS File Size in Vite

### DISCUSSION THREAD

**Discussion by harshitacodes** [vitejs/vite-plugin-react#341] [Q&A]
I'm trying to understand how my index.css is being generated. I know that Vite uses esbuild to build the file chunks. However, I want to reduce the size of index.css by removing some unused CSS during the build process. The current size of index.css is quite large, causing it to block other elements from loading and significantly impacting page performance.

Any suggestions on how to achieve this?

<img width="584" alt="Screenshot 2024-06-24 at 5 05 28 PM" src="https://github.com/vitejs/vite-plugin-react/assets/55912888/3ddd4d8a-7312-4ecb-8507-2ddfe1f9bc3f">

**Comment by SsomsakTH** [comment]
> I'm trying to understand how my index.css is being generated. I know that Vite uses esbuild to build the file chunks. However, I want to reduce the size of index.css by removing some unused CSS during the build process. The current size of index.css is quite large, causing it to block other elements from loading and significantly impacting page performance.
> 
> Any suggestions on how to achieve this?
> 
> 
> <img width="584" alt="Screenshot 2024-06-24 at 5 05 28 PM" src="https://github.com/vitejs/vite-plugin-react/assets/55912888/3ddd4d8a-7312-4ecb-8507-2ddfe1f9bc3f">
> 
> 
>
